gapis/resolve/synchronization_data.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resolve
+
+import (
+	"context"
+
+	"github.com/google/gapid/gapis/capture"
+	"github.com/google/gapid/gapis/gfxapi"
+)
+
+// Resolve builds a ResolvedResources object for all of the resources
+// at the path r.After
+func (r *SynchronizationResolvable) Resolve(ctx context.Context) (interface{}, error) {
```

Comment:
If this is just used inside the vulkan package, then you can move the resolvable into there.
If you can do that, do you still need `ResolveSynchronization` on the API interface?